File: eCinema.Web.API/ClassLibrary1/Entities/Hall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCInema.Models.Entities
{
    public class Hall
    {
        public int Id { get; set; }
        public int NoOfHall { get; set; }
        public int NumberOfSeats { get; set; }
        public int NumberOfRows { get; set; }
        public int NumberOfColumns { get; set; }
        public ICollection<Seat> Seats { get; set; }
    }
}

File: eCinema.Web.API/ecinemamobile/lib/models/Halls/hall.dart
import 'dart:core';
import 'dart:core';
import 'package:json_annotation/json_annotation.dart';

import '../Seats/seat.dart';
part 'hall.g.dart';

@JsonSerializable()
class Hall {
  int? noOfHall;
  int? numberOfSeats;
  int? numberOfRows;
  int? numberOfColumns;
  List<Seat>? seats;

  Hall(this.noOfHall, this.numberOfSeats, this.numberOfRows,
      this.numberOfColumns,
      [this.seats]);
  factory Hall.fromJson(Map<String, dynamic> json) => _$HallFromJson(json);

  Map<String, dynamic> HalltoJson() => _$HallToJson(this);
}
